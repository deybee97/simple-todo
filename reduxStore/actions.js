import login from "../api/login"

export const UPDATE_TODO ='UPDATE_TODO'

export const UPDATE_USER =  'UPDATE_USER'

export const LOG_IN_SENT = 'LOG_IN_SENT'
export const LOG_IN_FULFILLED = 'LOG_IN_FULFILLED'
export const LOG_IN_REJECTED = 'LOG_IN_REJECTED'


export const todoAction = (update) => ({
    type: UPDATE_TODO,
    payload: update,
  
})

export const userAction = (update) => ({
    type: UPDATE_USER,
    payload: update
})

export const loginUser =  (email, password, loginFn = login) => async (dispatch)  => {

    dispatch({type: LOG_IN_SENT})
    try {

      const token =  await loginFn(email, password)
        dispatch({type:LOG_IN_FULFILLED, payload:{token}})

    } catch (e) {
        dispatch({type:LOG_IN_REJECTED, payload:{errMsg: e.message}})
    }
    
    
} 






