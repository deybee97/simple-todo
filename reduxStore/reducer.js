import { combineReducers} from 'redux'

import { UPDATE_TODO, UPDATE_USER, LOG_IN_FULFILLED, LOG_IN_REJECTED, LOG_IN_SENT} from './actions'


const todoReducer = (state = [], action) => {

    switch(action.type){
    
        case UPDATE_TODO: return [ ...state, action.payload]
        default: return state
       
    }

}

const userReducer = (state={}, action) => {

    switch(action.type){
        case UPDATE_USER: return { ...state, ...action.payload}
        case LOG_IN_FULFILLED: return {...state, ...action.payload}
        default: return state
    }
}


export default combineReducers({
    todos: todoReducer,
    user: userReducer
})

