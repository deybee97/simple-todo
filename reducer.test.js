import reducer from "./reduxStore/reducer.js"
import * as action from "./reduxStore/actions.js"


const DEFAULT_STATE = {
    user:{},
    todos:[]
}

describe("contect reducer",()=>{

    it('sucessfully add new todo',()=>{
        expect(reducer(DEFAULT_STATE, action.todoAction("do this"))).toMatchSnapshot()
    })
    it('sucessfully update user object',()=>{
        expect(reducer(DEFAULT_STATE, action.userAction({token: "otARealToken"}))).toMatchSnapshot()
    })

})