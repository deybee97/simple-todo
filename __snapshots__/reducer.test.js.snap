// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contect reducer sucessfully add new todo 1`] = `
{
  "todos": [
    "do this",
  ],
  "user": {},
}
`;

exports[`contect reducer sucessfully update user object 1`] = `
{
  "todos": [],
  "user": {
    "token": "otARealToken",
  },
}
`;
